x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -185,8 +186,7 @@ public void getSnapshotInfo(GetSnapshotInfoContext context) {
             .get(ccrSettings.getRecoveryActionTimeout());
         Metadata metadata = response.getState().metadata();
         ImmutableOpenMap<String, IndexMetadata> indicesMap = metadata.indices();
-        ArrayList<String> indices = new ArrayList<>(indicesMap.size());
-        indicesMap.keysIt().forEachRemaining(indices::add);
+        List<String> indices = indicesMap.keysStream().collect(Collectors.toList());
```

Comment:
Using the proposal that Ryan made, this can turn into `List.copyOf(indicesMap.keySet())` from java 10 and until then `new ArrayList<>(indicesMap.keySet())`. That seems better than stream collecting it.